File: iOS OC Example/iOS OC Example/InterstitialViewController.m
//
//  InterstitialViewController.m
//  iOS OC Example
//
//  Created by M on 2019/5/28.
//  Copyright © 2019 1tu1shu. All rights reserved.
//

#import "InterstitialViewController.h"
#import <AdTimingSDK/AdTimingSDK.h>

@interface InterstitialViewController ()<AdTimingInterstitialAdDelegate>

@property(nonatomic,strong)UIButton *loadBtn;
@property(nonatomic,strong)UIButton *showBtn;
@property(nonatomic,strong)UILabel *logLabel;

@end

@implementation InterstitialViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"Interstitial";
    self.view.backgroundColor = [UIColor whiteColor];
    
    self.showBtn = [UIButton buttonWithType:UIButtonTypeSystem];
    self.showBtn.frame = CGRectMake(20, CGRectGetMaxY(self.loadBtn.frame)+20, 120, 30);
    [self.showBtn setTitle:@"Show" forState:UIControlStateNormal];
    self.showBtn.contentHorizontalAlignment = UIControlContentHorizontalAlignmentLeft;
    self.showBtn.backgroundColor = [UIColor whiteColor];
    self.showBtn.titleLabel.font = [UIFont boldSystemFontOfSize:18];
    self.showBtn.layer.cornerRadius = 5;
    [self.showBtn addTarget:self action:@selector(showBtnDidClicked) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:self.showBtn];
    
    self.logLabel = [[UILabel alloc] initWithFrame:CGRectMake(20, CGRectGetMaxY(self.showBtn.frame)+20, self.view.frame.size.width-40, 30)];
    self.logLabel.text = @"Log";
    self.logLabel.font = [UIFont systemFontOfSize:15];
    [self.view addSubview:self.logLabel];
    
    // Setting Interstitial Delegate
    [[AdTimingInterstitialAd sharedInstance] addDelegate:self];
    
}

-(void)showBtnDidClicked
{
    // ShowInterstitial
    if ([[AdTimingInterstitialAd sharedInstance] isReady]) {
        [[AdTimingInterstitialAd sharedInstance] showWithViewController:self scene:@"YOUR_SCENE_NAME"];
    }
}

/// Invoked when interstitial video is available.
/// You can then show the video by calling show method.
- (void)adtimingInterstitialChangedAvailability:(BOOL)available{
    if(available){
        NSLog(@"InterstitialAd Did Load");
    }
}

/// Sent immediately when a interstitial video starts to play.
- (void)adtimingInterstitialDidShow:(AdTimingScene *)scene{
    NSLog(@"InterstitialAd Start Play");
}

/// Sent immediately when a interstitial video is opened.
- (void)adtimingInterstitialDidOpen:(AdTimingScene *)scene{
    NSLog(@"InterstitialAd Start Play");
}

/// Sent after a interstitial video has been clicked.
- (void)adtimingInterstitialDidClick:(AdTimingScene *)scene{
    NSLog(@"InterstitialAd Did Click");
}

/// Sent after a interstitial video has been closed.
- (void)adtimingInterstitialDidClose:(AdTimingScene *)scene{
    NSLog(@"InterstitialAd Did Close");
}

/// Sent after a interstitial video has failed to play.
- (void)adtimingInterstitialDidFailToShow:(AdTimingScene *)scene withError:(NSError *)error{
    NSLog(@"InterstitialAd failed to play");
}

@end

File: iOS OC Example/iOS Swift Example/InterstitialViewController.swift
//
//  InterstitialViewController.swift
//  iOS Swift Example
//
//  Created by M on 2019/5/28.
//  Copyright © 2019 1tu1shu. All rights reserved.
//

import UIKit
import AdTimingSDK
class InterstitialViewController: UIViewController, AdTimingInterstitialAdDelegate {
    
    var logLabel = UILabel()
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.title = "Interstitial"
        self.view.backgroundColor = UIColor.white
            
        let showBtn:UIButton = UIButton(frame: CGRect(x: 20, y: 120, width: 120, height: 30))
        showBtn.contentHorizontalAlignment = UIControl.ContentHorizontalAlignment.left
        showBtn.backgroundColor = UIColor.white
        showBtn.setTitleColor(UIColor.blue, for: .normal)
        showBtn.setTitleColor(UIColor.gray, for: .highlighted)
        showBtn.setTitleColor(UIColor.gray, for: .selected)
        showBtn.titleLabel?.font = UIFont.systemFont(ofSize: 17)
        showBtn.setTitle("Show", for: .normal)
        showBtn.addTarget(self, action: #selector(showBtnDidClicked), for: .touchUpInside)
        self.view.addSubview(showBtn)
        
        logLabel = UILabel(frame: CGRect(x: 20, y: showBtn.frame.maxY+20, width: 120, height: 30))
        logLabel.text = "Log"
        logLabel.font = UIFont.systemFont(ofSize: 15)
        self.view.addSubview(logLabel)
    }
    
    @objc func showBtnDidClicked(){
        if AdTimingInterstitialAd.sharedInstance().isReady() {
            AdTimingInterstitialAd.sharedInstance().show(with: self, scene: "")
        }
    }
    
    func adtimingInterstitialChangedAvailability(_ available: Bool) {
        if available {
           print("Ad is Available")
        }
    }
    
    func adtimingInterstitialDidOpen(_ scene: AdTimingScene) {
        print("Ad is Open")
    }
    
    func adtimingInterstitialDidShow(_ scene: AdTimingScene) {
        print("AdStartPlay")
    }
    
    func adtimingInterstitialDidClick(_ scene: AdTimingScene) {
        print("AdDidClick")
    }
        
    func adtimingInterstitialDidClose(_ scene: AdTimingScene) {
        print("AdDidClose")
    }
    
    func adtimingInterstitialDidFail(toShow scene: AdTimingScene, withError error: Error) {
        print("AdFailedToPlay")
    }
}
